diff --git a/src/components/home/elements/Record.vue b/src/components/home/elements/Record.vue
index 12f4272ce735ca2922ed084c56beb2369de18dcc..bd519e9c35b13ec30647d7cf011fc4662a75e33b 100644
--- a/src/components/home/elements/Record.vue
+++ b/src/components/home/elements/Record.vue
@@ -159,7 +159,6 @@
 
 <script>
 import TxInfoModal from './TxInfoModal'
-import Vue from 'vue'
 import base58 from '@/libs/base58'
 import converters from '@/libs/converters'
 import crypto from '@/utils/crypto'
@@ -225,21 +224,6 @@ export default {
         }
     },
     computed: {
-        defaultAddress() {
-            return Vue.ls.get('address')
-        },
-        userInfo() {
-            return JSON.parse(window.localStorage.getItem(this.defaultAddress))
-        },
-        getDevice() {
-            if (this.userInfo && this.userInfo.coldAddresses) {
-                var object = JSON.parse(this.userInfo.coldAddresses)
-                if (object[this.address] && object[this.address].hasOwnProperty('device')) {
-                    return object[this.address].device
-                }
-            }
-            return ''
-        },
         selfSend() {
             if (this.txRecord.recipient === this.address && this.txRecord.SelfSend === true) {
                 return 'selfSend'
@@ -420,11 +404,7 @@ export default {
             this.$root.$emit('bv::show::modal', 'txInfoModal_' + this.transType + this.txRecord.id + this.selfSend)
         },
         cancelLeasing() {
-            if (this.getDevice === 'Ledger') {
-                alert('This feature is not supported')
-            } else {
-                this.$root.$emit('bv::show::modal', 'cancelLeaseModal_' + this.txRecord.id)
-            }
+            this.$root.$emit('bv::show::modal', 'cancelLeaseModal_' + this.txRecord.id)
         },
         showDetails(cancelTime) {
             this.cancelTime = cancelTime
diff --git a/src/components/home/modals/CancelLease.vue b/src/components/home/modals/CancelLease.vue
index 3a111664415694734986dedc395e92ae7dda3e27..591ee5d29491d2e86e8dc494f4e9a2757fd88dbb 100644
--- a/src/components/home/modals/CancelLease.vue
+++ b/src/components/home/modals/CancelLease.vue
@@ -59,12 +59,33 @@
     <b-container v-else-if="page==='success'">
       <CancelSuccess @show-details="showDetails"></CancelSuccess>
     </b-container>
-    <b-container v-else-if="page==='cold'">
+    <b-container v-else-if="page==='cold' && getDevice==='Ledger'"
+                 class="text-left">
       <ColdSignature :data-object="dataObject"
                      @get-signature="getSignature"
                      @prev-page="prevPage"
                      :tx-type="txType"></ColdSignature>
     </b-container>
+    <b-container v-else-if="page==='cold' && getDevice!=='Ledger'"
+                 class="text-left">
+      <LedgerConfirm :address="address"
+                     :recipient="recipient"
+                     :amount=amount
+                     :fee="fee"
+                     :attachment="attachment"
+                     :tx-type="'Cancel lease'"></LedgerConfirm>
+      <b-row class="row">
+        <b-col class="col-back">
+          <b-button
+            class="btn-back"
+            block
+            variant="light"
+            size="lg"
+            @click="prevPage">Back
+          </b-button>
+        </b-col>
+      </b-row>
+    </b-container>
   </b-modal>
 </template>
 
@@ -79,9 +100,10 @@ import seedLib from '@/libs/seed'
 import Vue from 'vue'
 import browser from '@/utils/browser'
 import BigNumber from 'bignumber.js'
+import LedgerConfirm from './LedgerConfirm'
 export default {
     name: 'CancelLease',
-    components: { TxInfoModal, CancelSuccess, ColdSignature, Confirm },
+    components: { TxInfoModal, CancelSuccess, ColdSignature, Confirm, LedgerConfirm },
     data: function() {
         return {
             errorMessage: '',
@@ -91,7 +113,8 @@ export default {
             sendError: false,
             signed: false,
             txType: CANCEL_LEASE_TX,
-            hasConfirmed: false
+            hasConfirmed: false,
+            attachment: ''
         }
     },
     props: {
@@ -179,6 +202,15 @@ export default {
         },
         seedPhrase() {
             return seedLib.decryptSeedPhrase(this.secretInfo.encrSeed, Vue.ls.get('pwd'))
+        },
+        getDevice() {
+            if (this.userInfo && this.userInfo.coldAddresses) {
+                var object = JSON.parse(this.userInfo.coldAddresses)
+                if (object[this.address] && object[this.address].hasOwnProperty('device')) {
+                    return object[this.address].device
+                }
+            }
+            return ''
         }
     },
     methods: {
@@ -336,4 +368,14 @@ export default {
 .col-rit {
     padding-left: 10px;
 }
+.col-back {
+    padding-left: 10px;
+    margin-top: -70px;
+    margin-right: 230px;
+    margin-left: 5px;
+}
+.row {
+    margin-top: 26px;
+    margin-bottom: 10px;
+}
 </style>
